### DISCUSSION TITLE
Nesting Select Component

### DISCUSSION THREAD

**Discussion by ousid** [filamentphp/filament#6302] [Help] [answered]
I have been using filament for quite a while now, and I got a situation to display nested result with a `Select` component, but I didn't find any solution so far.

### Steps to produce

I have the following schema:

```
- Group 1
  - Value 01
  - Value 02
  - ...
- Group 2
  - Value 01
  - Value 02
  - ...
- ...
```

Using the following implementation:

```php
Forms\Components\Select::make('choices')
   ->options($options),
```

## Problems

I got one of the following error:

```
array_key_exists(): Argument #1 ($key) must be a valid array offset type
```

Or 
```
Filament\Forms\Components\Select::isOptionDisabled(): Argument #2 ($label) must be of type string, array given
```

When I take a look at the source code, I found the `Select` method that check the `array` and transform the result into `key => value`, and I cannot use something like `['key' => ['value']]`

Also, when I use `searchable` method with the `Select` Component, in this situation, the result back as undefined objects, something like

```
[Object, Object, Object, ...]
[Object, Object, Object, ...]
```

Is there any way to achieve the nested result in the current `Select` component implementation?

__PS__: I'm saving the [Independent Selects](https://filamentphp.com/docs/2.x/forms/fields#dependant-selects) as my last bullet, if I cannot find any solution for the time being.

Thank you in advance!

**Comment by danharrin** [comment, marked as answer]
Yeah we do not support grouping on the select component. It would be a very complex feature that I do not think I want to support yet, since it would drastically change the maintainability of the component. I would suggest creating a simple custom field with a `<select>` `<optgroup>`

**Reply by ousid** [reply]
Yeah, I thought so. I decided to go with the independent select. It serves the purpose well, and it's a better from the `UX` perspective in my case.

Thanks for your response, Dan!
